.github/workflows/docs.yml

```diff
@@ -18,13 +18,64 @@ name: "Documentation check"
 on:
 - pull_request
 
+env:
+  DEPS_BRANCH: bot/stable-deps-update
+
 jobs:
   sphinx:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v3
     - uses: PennyLaneAI/sphinx-action@master
       with:
         docs-folder: "doc/"
-        pre-build-command: "apt install -y graphviz && pip3 install -r doc/requirements.txt && pip3 install . pip3 install openfermionpyscf && pip3 install aiohttp fsspec h5py"
+        pre-build-command: "apt install -y graphviz && pip3 install -r doc/requirements.txt && pip3 install . && pip3 install openfermionpyscf && pip3 install aiohttp fsspec h5py && pip freeze | grep -v 'file:///' > .github/stable/doc.txt.tmp"
         build-command: "sphinx-build -b html . _build -W --keep-going"
+
+    - name: Prepare local repo
+      run: |
+        git fetch
+        git config user.name "GitHub Actions Bot"
+        git config user.email "<>"
+        if git ls-remote --exit-code origin "refs/heads/${{ env.DEPS_BRANCH }}"; then
+          git checkout "${{ env.DEPS_BRANCH }}"
+        else
+          git checkout master
+          git pull
+          git checkout -b "${{ env.DEPS_BRANCH }}"
+        fi
+        mv -f .github/stable/doc.txt.tmp .github/stable/doc.txt
+
+    - name: Determine if changes have been made
+      id: changed
+      run: |
+        echo "has_changes=$(git status --porcelain | wc -l | awk '{print $1}')" >> $GITHUB_OUTPUT
+
+    - name: Stage changes
+      if: steps.changed.outputs.has_changes != '0'
+      run: |
+        git add .github/stable/doc.txt
+        git commit -m "Update stable docs dependencies"
+        git push -f --set-upstream origin "${{ env.DEPS_BRANCH }}"
+
+    # Create PR to master
+    - name: Create pull request
+      if: steps.changed.outputs.has_changes != '0'
+      uses: repo-sync/pull-request@v2
+      with:
+        source_branch: "${{ env.DEPS_BRANCH }}"
+        destination_branch: "master"
+        github_token: "${{ secrets.GITHUB_TOKEN }}"
+        pr_title: "Update stable dependency files"
+        pr_body: |
+          Automatic update of stable requirement files to snapshot valid python environments.
+
+          Because bots are not able to trigger CI on their own, please do so by pushing an empty commit to this branch using the following command:
+
+          ```
+          git commit --allow-empty -m 'trigger ci'
+          ```
+
+          Alternatively, wait for this branch to be out-of-date with master, then just use the "Update branch" button!
+        pr_allow_empty: false
```

Comment:
# Question

Does this mean we don't actually need `if: steps.changed.outputs.has_changes != '0'` above?

Reply:
probably but I'm happy with the double-safeguard. just skip this step if we don't need it, and if somehow there are no commits despite us expecting one, we'll get a nice error here

Reply:
Gotcha. I was mainly asking to make sure I understood things correctly; this sounds good to me.